ckanapi/common.py

```diff
@@ -40,7 +40,7 @@ def __getattr__(self, name):
         def action(**kwargs):
             files = {}
             for k, v in kwargs.items():
-                if hasattr(v, 'read'):
+                if hasattr(v, 'read') or (isinstance(v, (list, tuple)) and len(v) >= 2 and hasattr(v[1], 'read')):
```

Comment:
You're probably right about the lists.

As for trailing elements, according to http://docs.python-requests.org/en/latest/user/quickstart/#post-a-multipart-encoded-file 
one can sepcify content-type and headers, ie.
`files = {'file': ('report.xls', open('report.xls', 'rb'), 'application/vnd.ms-excel', {'Expires': '0'})}`